packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -419,18 +596,23 @@ export class ModularChangeFamily
 			revisionMetadata,
 		);
 
-		const nodeId = id1 ?? id2 ?? fail("Should not compose two undefined node IDs");
-		setInNestedMap(
-			composedNodeChanges,
-			nodeId.revision,
-			nodeId.localId,
-			composedNodeChangeset,
-		);
+		setInChangeAtomIdMap(composedNodes, id1, composedNodeChangeset);
+
+		if (!areEqualChangeAtomIds(id1, id2)) {
+			deleteFromNestedMap(composedNodes, id2.revision, id2.localId);
+			deleteFromNestedMap(composedNodeToParent, id2.revision, id2.localId);
+			setInChangeAtomIdMap(nodeAliases, id2, id1);
+
+			// We need to delete id1 to avoid forming a cycle in case id1 already had an alias.
+			deleteFromNestedMap(nodeAliases, id1.revision, id1.localId);
```

Comment:
I'm confused: I can see a risk of cycle if we had an id1 => id2 alias already, is that even possible?
More importantly, if we have an id1 => idX alias where idX !=== id2, don't we rely on that alias chain later on?